### DISCUSSION TITLE
FUNC_GODOT IS HERE!

### DISCUSSION THREAD

**Discussion by RhapsodyInGeek** [func-godot/func_godot_plugin#1] [Announcements]
### WE ARE PROUD AND HAPPY TO ANNOUNCE THE FIRST RELEASE OF *FUNC_GODOT*, A QUAKE AND HALF-LIFE MAP IMPORTER FOR GODOT 4!

FuncGodot is designed to be as map editor agnostic as possible: so far it has tested well with TrenchBroom, J.A.C.K., and NetRadiant Custom. Support for more map editors is possible with FuncGodot being less dependent on TrenchBroom, while still providing direct support for it.

Forked from the Godot 4 port of Qodot, the design of that plugin was reworked based upon the Qodot Team's deeper research into the map format and a desire for a more focused and flexible implementation.

Entity definition and generation has also been reworked, with more flexibility in applying properties and referencing other generated nodes in the map build process, as well as increased customization and compatibility options for brush to mesh generation.

FuncGodot is designed at its core to be source control friendly, providing local machine configurations that work neatly alongside tracked FGD and game configuration resources.

On top of all of that are new quality of life features that make the FuncGodot Map building processes that much smoother and easier to follow. Every single modifiable property in every single resource is fully commented, and there exists full documentation of all of FuncGodot's features available both online and offline.

### REGARDING QODOT!

Some of Qodot Community members may have some questions about FuncGodot: Why a whole new plugin? Why the rebranding? Why not just update Qodot? What does the future of Qodot look like with the release of FuncGodot?

There was a strong desire to rewrite Qodot from the ground up, due to some longstanding internal issues with the build process and its maintainability. There were also some questionably designed implementations that relied on either misunderstood knowledge or assumptions in the Quake map format and how TrenchBroom works with it. We knew the rewrite would end up completely breaking people's projects, something we had managed to avoid with every Qodot update published. We were hitting a wall with Qodot's tech debt, but we didn't want to break people's projects. It's difficult to change gears when you're far along enough in a production.

Another problem we were having is the question of transferable administrative access to both the Qodot Discord Server and Qodot GitHub Organization. While we are all maintainers on the Qodot 4 Repository, we are limited in how we can serve and maintain the Qodot Community without this access and as such cannot make any guarantees on the persistence or maintenance of the Qodot social and code channels.

So the answer to us seemed to be to branch off. Out of respect for Qodot, to avoid confusion between the two surface-level-similar plugins and to embrace the pillar of map editor agnosticism, we decided to rebrand the extremely heavily modified fork of Qodot to FuncGodot.

So what does the future of Qodot look like? Well, the Qodot Team will be winding down on active maintenance. Don't expect new features unless they can be guaranteed non-breaking and within Qodot's scope. Bug fixes are welcome, but we will no longer be actively working on it. Qodot 4 2024.1 is likely to be the last AssetLib and Release version of Qodot, adding needed support for Version 8 of the TrenchBroom GameConfig.

While we do not have the final say in what happens to the Qodot repository, organization, and Discord server, our intention is to leave them accessible for other users who wish to continue to use them.

### FINALLY, FUNC_GODOT

After months of simultaneously unfocused and hyper-focused work, we're extremely excited to be releasing FuncGodot to the game dev community at large. You can download the plugin through the [**FuncGodot GitHub Repository**](https://github.com/func-godot/func_godot_plugin) or through the Godot 4 Asset Library.

Support for FuncGodot can be found through both the [**full online documentation**](https://func-godot.github.io/func_godot_docs) and on the new [**FuncGodot Discord Server**](https://discord.gg/zje67FcC6N).

***- FuncGodot Team***

**Comment by roboboredom** [comment]
Doing the lords work fr!

**Comment by therealstrikeman** [comment]
does it support godot 4.2.2 version

**Reply by RhapsodyInGeek** [reply]
Yes.

**Comment by 13Character5** [comment]
This will be perfect because now i can ditch the .NET version of Godot 4.2.2. Qodot is the only reason why i use godot with mono support
